File: src/dev/wirlie/bungeecord/glist/servers/ServerGroup.java
package dev.wirlie.bungeecord.glist.servers;

import net.md_5.bungee.api.config.ServerInfo;
import net.md_5.bungee.api.connection.ProxiedPlayer;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class ServerGroup implements ServerInfoProvider {

    private final String id;
    private List<ServerInfo> servers = new ArrayList<>();

    public ServerGroup(String id) {
        this.id = id;
    }

    public void setServers(List<ServerInfo> servers) {
        this.servers = servers;
    }

    public List<ServerInfo> getServers() {
        return servers;
    }

    @Override
    public String getId() {
        return id;
    }

    @Override
    public int getPlayerCount() {
        return servers.stream().mapToInt(s -> s.getPlayers().size()).sum();
    }

    @Override
    public List<ProxiedPlayer> getPlayers() {
        return servers.stream().flatMap(s -> s.getPlayers().stream()).collect(Collectors.toList());
    }

    @Override
    public String getDisplayName() {
        return id + " (" + servers.stream().map(ServerInfo::getName).collect(Collectors.joining(", ")) + ")";
    }
}

File: Platform-BungeeCord/src/main/kotlin/dev/wirlie/bungeecord/glist/servers/ServerGroup.kt
package dev.wirlie.bungeecord.glist.servers

import net.md_5.bungee.api.config.ServerInfo
import net.md_5.bungee.api.connection.ProxiedPlayer
import java.util.stream.Collectors

class ServerGroup(override val id: String) : ServerInfoProvider {

    var servers: List<ServerInfo> = ArrayList()

    override val playerCount: Int
        get() = servers.stream().mapToInt { s: ServerInfo -> s.players.size }.sum()

    override val players: List<ProxiedPlayer>
        get() = servers.stream().flatMap { s: ServerInfo -> s.players.stream() }.collect(Collectors.toList())

    override val displayName: String
        get() = id + " (" + servers.stream().map { obj: ServerInfo -> obj.name }.collect(Collectors.joining(", ")) + ")"

}
